src/test/ui/lint/lint-exceeding-bitshifts.rs

```diff
@@ -1,3 +1,5 @@
+// compile-flags: -O
```

Comment:
Shouldn't this warn in debug mode too?

Reply:
@oli-obk Just in case this has been overlooked: we should always run the constant-folding pass and *only commit the results* when optimizations are enabled (although maybe we shouldn't care about optimization level at all?).

Reply:
That's what we do, the differences arise because the MIR differs and thus different lints are reported